modules/node-shared/src/main/scala/io/constellationnetwork/node/shared/infrastructure/consensus/ConsensusResources.scala

```diff
@@ -21,19 +21,34 @@ case class ConsensusResources[A, Kind](
   withdrawalsMap: Map[PeerId, Kind],
   ackKinds: Set[Kind],
   artifacts: Map[Hash, A],
-  updatedAt: FiniteDuration
+  updatedAt: FiniteDuration,
+  createdAt: FiniteDuration,
+  facilities: Option[FiniteDuration],
+  facilitiesLatestUnique: Option[FiniteDuration],
+  proposals: Option[FiniteDuration],
+  proposalsLatestUnique: Option[FiniteDuration],
+  signatures: Option[FiniteDuration],
+  signaturesLatestUnique: Option[FiniteDuration]
 )
```

Comment:
These represent first event occurrence and latest unique change, we probably don't need so many but its useful for event log tracing.